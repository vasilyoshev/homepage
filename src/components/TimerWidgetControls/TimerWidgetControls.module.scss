.buttonsContainer {
  justify-content: space-between;

  Button {
    margin-left: 8px;
  }
}
