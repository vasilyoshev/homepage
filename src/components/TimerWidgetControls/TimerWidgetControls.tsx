import React from 'react';
import { TimerWidgetControlsProps } from 'interfaces';
import Button from '@material-ui/core/Button';
import styles from './TimerWidgetControls.module.scss';

export const TimerWidgetControls: React.FC<TimerWidgetControlsProps> = ({ timerOn, time, startTimer, stopTimer, resetTimer, resumeTimer }) => {
  return (
    <section className={styles.buttons}>
      {!timerOn && time === 0 && (
        <Button variant="outlined" color="primary" onClick={startTimer}>Start</Button>
      )}
      {timerOn && <Button variant="outlined" color="secondary" onClick={stopTimer}>Stop</Button>}
      {!timerOn && time > 0 && (
        <Button variant="outlined" color="secondary" onClick={resetTimer}>Reset</Button>
      )}
      {!timerOn && time > 0 && (
        <Button variant="outlined" color="primary" onClick={resumeTimer}>Resume</Button>
      )}
    </section>
  );
};
