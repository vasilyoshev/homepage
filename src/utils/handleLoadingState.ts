import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
