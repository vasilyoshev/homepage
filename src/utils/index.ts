export * from './handleLoadingState';
export * from './testUtils';
