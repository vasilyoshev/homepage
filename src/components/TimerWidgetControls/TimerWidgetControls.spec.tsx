import { render } from 'utils';
import { TimerWidgetControls } from './TimerWidgetControls';

describe('TimerWidgetControls', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<TimerWidgetControls
      timerOn={true}
      time={0}
      startTimer={jest.fn()}
      stopTimer={jest.fn()}
      resetTimer={jest.fn()}
      resumeTimer={jest.fn()}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
