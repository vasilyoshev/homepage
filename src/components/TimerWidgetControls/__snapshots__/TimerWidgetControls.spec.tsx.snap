// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerWidgetControls renders successfully 1`] = `
<DocumentFragment>
  <section
    class="buttonsContainer"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Stop
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
